app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+          checkYourAnswersHandler(c, fillingOutReturn, s)
+        case _ =>
+          logger.warn("error processing check your answers")
+          errorHandler.errorResult()
+      }
+    }
+  }
+
+  def checkYourAnswersSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withUploadSupportingEvidenceAnswers(request) { (_, _, _, fillingOutReturn, answers) =>
+        val updatedAnswers: UploadSupportingEvidenceAnswers = answers match {
+          case IncompleteUploadSupportingEvidenceAnswers(None, _) =>
+            CompleteUploadSupportingEvidenceAnswers(
+              doYouWantToUploadSupportingEvidence = false,
+              List.empty
+            ) // shut the compiler up - this can never happen actually
+          case IncompleteUploadSupportingEvidenceAnswers(
+              Some(doYouWantToUploadSupportingDocumentAnswer),
+              supportingDocuments
+              ) =>
+            CompleteUploadSupportingEvidenceAnswers(doYouWantToUploadSupportingDocumentAnswer, supportingDocuments)
+          case CompleteUploadSupportingEvidenceAnswers(
+              doYouWantToUploadSupportingDocumentAnswer,
+              supportingDocuments
+              ) =>
+            CompleteUploadSupportingEvidenceAnswers(doYouWantToUploadSupportingDocumentAnswer, supportingDocuments)
+        }
+        val newDraftReturn = fillingOutReturn.draftReturn match {
+          case s @ DraftSingleDisposalReturn(
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _
+              ) =>
+            s.copy(uploadSupportingDocuments = Some(updatedAnswers))
+          case m @ DraftMultipleDisposalsReturn(
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _
+              ) =>
+            m.copy(uploadSupportingDocuments = Some(updatedAnswers))
+        }
+
+        val result = for {
+
+          _ <- returnsService
+                .storeDraftReturn(
+                  newDraftReturn,
+                  fillingOutReturn.subscribedDetails.cgtReference,
+                  fillingOutReturn.agentReferenceNumber
+                )
+          _ <- EitherT(
+                updateSession(sessionStore, request)(
+                  _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = newDraftReturn)))
+                )
+              )
+        } yield ()
+
+        result.fold({ e =>
+          logger.warn("Could not update session", e)
+          errorHandler.errorResult()
+        }, _ => Redirect(controllers.returns.routes.TaskListController.taskList()))
+      }
+    }
+
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements", "org.wartremover.warts.Any"))
```

Comment:
`NonUnitStatements` is usually a sign that something is not being handled properly - why do we need to ignore the warning here?

Reply:
Removed - changed the implementation but forgot to take this out.